frontend-react/src/components/USLink.tsx

```diff
@@ -1,38 +1,103 @@
-import React, { PropsWithChildren } from "react";
+import React, { AnchorHTMLAttributes } from "react";
 import { Link, NavLink } from "react-router-dom";
+import classnames from "classnames";
 
-// Known issue with the `PropsWithChildren generic in React 18,
-// so I wrote this in a way that we can just remove `<any>` and be
-// okay if we update our React version.
-interface USNavLinkProps extends PropsWithChildren<any> {
-    href: string;
+/** React.PropsWithChildren has known issues with generic extension in React 18,
+ * so rather than using it here, we are using our own definition of child types.
+ * One less headache when updating to React 18 in the future! */
+interface CustomLinkProps {
+    children: React.ReactNode;
+    anchor?: boolean;
     className?: string;
     activeClassName?: string;
 }
-type USLinkProps = Omit<USNavLinkProps, "activeClassName">;
+type USLinkProps = AnchorHTMLAttributes<{}> &
+    Omit<CustomLinkProps, "activeClassName">;
+type USNavLinkProps = Pick<AnchorHTMLAttributes<{}>, "href"> & CustomLinkProps;
 
-/** A single link for rendering standard links */
-export const USLink = ({ children, href, className }: USLinkProps) => (
-    <Link to={href} className={`usa-link ${className}`}>
-        {children}
-    </Link>
-);
+/** A single link for rendering standard links. Uses a `Link` by default
+ * but adding `anchor` will make this a generic anchor tag.
+ * @example
+ * <USLink href="/page">To Page</USLink> // uses <Link> from react-router-dom
+ * <USLink anchor href="#this-section-on-my-page">To Section</USLink> // uses <a>
+ * */
+export const USLink = ({
+    anchor = false,
+    children,
+    className,
+    href,
+    ...anchorHTMLAttributes
+}: USLinkProps) => {
+    return !anchor ? (
```

Comment:
Small thing, but to reduce a bit of redundancy/future maintenance, we could just toggle the tag:

```jsx
const Tag = anchor ? 'a' : Link;

return (
  <Tag to={ .... }>{ children }</Tag>  
);
```

Reply:
YES! I was trying to Google how to do this because I'd seen it before in some React code but couldn't find anyone answering it. Will try to implement it this way

Reply:
So, in the screenshot below, I tried the above implementation and we get an issue with testing where the tag rendered is _not_ recognized as a link if it's an `anchor` tag. That gives me some concern for accessibility even though it's clear that we're rendering it as `<a>` in the DOM.

<img width="1442" alt="Screen Shot 2023-01-04 at 11 04 50 AM" src="https://user-images.githubusercontent.com/4022304/210597822-eabbd7b5-6b66-4cf6-bc09-3aa320bad747.png">

And then I realized, due to the `to` vs `href` difference between `Link` and `a`, we end up rendering links that don't link anywhere. Are you aware of a way to conditionally name that prop? If not, I think the current version might be our best bet.

<img width="433" alt="Screen Shot 2023-01-04 at 11 10 41 AM" src="https://user-images.githubusercontent.com/4022304/210598940-35cdff6b-dd27-49ba-9abc-9f53b940694a.png">

Reply:
Hehehe oops, duh!  I think we could do something like this, though:

```jsx
const props = {
    ...otherProps,
    [anchor ? 'a' : 'to']: href
};

return <Tag { ...props } />;
```

Not a big deal, but I think it helps to cut down redundancy and potential for typos if we add more props 🤷

Reply:
I was trying to do this inline with JSX and it hated it; gonna try this! :)

Reply:
Bah humbug...still doesn't like it at the JSX level. Tried both inline and as the `toOrHref` var.

<img width="1022" alt="Screen Shot 2023-01-04 at 11 51 19 AM" src="https://user-images.githubusercontent.com/4022304/210607227-86870ac6-128c-4939-be3b-cf1bd07a5129.png">